lib/canvas_link_migrator/link_resolver.rb

```diff
@@ -219,13 +224,13 @@ def media_attachment_iframe_url(file_id, media_type = nil)
 
     def resolve_media_comment_data(node, rel_path)
       if (file = find_file_in_context(rel_path[/^[^?]+/])) # strip query string for this search
-        media_id = (file.media_object&.media_id || file.media_entry_id)
+        media_id = file["media_entry_id"]
```

Comment:
Do we know if this might break in certain instances in Canvas? My knowledge on this is shaky, but aren't some of them only connected via media_object.id (hence the need for `file.media_object&.media_id`). 

I know that anything just using the resource map won't have `file.media_object`, but we could just check that file implements that method first.

Reply:
In Canvas, the media object has the foreign key to the attachment, and attachments always just have media_entry_id.

It's possible if there was a race condition that the attachment might not be able to find the media object while notorious is still processing media, because it wouldn't have a media entry id.  And in that case, it's possible the media object _would_ have the media id, but I'm not super worried about that case.  Having said that, it wouldn't be too difficult to have both cases.

Reply:
Gotcha, okay, I'm cool with it as is.